src/shared/conf-parser.h

```diff
@@ -380,17 +373,14 @@ typedef enum ConfigParseStringFlags {
                 }                                                       \
                                                                         \
                 x = name##_from_string(rvalue);                         \
-                if (x < 0) {                                            \
-                        log_syntax(unit, LOG_WARNING, filename, line, x, \
-                                   msg ", ignoring: %s", rvalue);       \
-                        return 0;                                       \
-                }                                                       \
+                if (x < 0)                                              \
+                        return log_syntax_parse_error(unit, filename, line, x, lvalue, rvalue); \
                                                                         \
                 *i = x;                                                 \
                 return 1;                                               \
         }
 
-#define DEFINE_CONFIG_PARSE_ENUMV(function, name, type, invalid, msg)          \
+#define DEFINE_CONFIG_PARSE_ENUMV(function, name, type, invalid)               \
```

Comment:
BTW, new0 in this macro can be replaced with new() or newdup()

And realloc -> reallocarray?